service/worker/scheduler/workflow.go

```diff
@@ -1544,3 +1560,12 @@ func GetListInfoFromStartArgs(args *schedspb.StartScheduleArgs, now time.Time, s
 	s.State.LastProcessedTime = timestamppb.New(now)
 	return s.getListInfo(false)
 }
+
+func isUserScheduleError(err error) bool {
+	var appErr *temporal.ApplicationError
+	var workflowExecutionAlreadyStarted *serviceerror.WorkflowExecutionAlreadyStarted
+	if errors.As(err, &appErr) && errors.As(appErr.Unwrap(), &workflowExecutionAlreadyStarted) {
```

Comment:
could you check that this works without the explicit Unwrap? (i.e. like the first version)  errors.As calls Unwrap automatically so it shouldn't be necessary